python_modules/dagster/dagster/_config/structured_config.py

```diff
@@ -15,6 +20,46 @@ class Config(BaseModel):
     """
 
 
+class StructuredConfigResource(
+    ResourceDefinition,
+    Config,
+):
+    def __init__(self, **data: Any):
+        schema = infer_schema_from_config_class(self.__class__)
+
+        inner_resource_def = ResourceDefinition(self.resource_fn, schema)
+        configured_resource_def = inner_resource_def.configured(
+            config_dictionary_from_values(
+                data,
+                schema,
+            ),
+        )
+
+        Config.__init__(self, **data)
+        ResourceDefinition.__init__(
+            self,
+            resource_fn=self.resource_fn,
+            config_schema=configured_resource_def.config_schema,
+            description=self.__doc__,
+        )
+
+    def __setattr__(self, name: str, value: Any):
+        # This is a hack to allow us to set attributes on the class that are not part of the
+        # config schema. Pydantic will normally raise an error if you try to set an attribute
+        # that is not part of the schema.
+
+        if name.startswith("_"):
+            object.__setattr__(self, name, value)
+            return
+
+        return super().__setattr__(name, value)
+
+    def resource_fn(self, context) -> Any:  # pylint: disable=unused-argument
```

Comment:
rename `context` to `_` to avoid pylint pragma

Reply:
I was leaving it as `context` so the kwarg matches up in case a child class chooses to override `resource_fn`, e.g. a case where a user wants to wrap a function-style resource.

Reply:
ah I see. ok

Reply:
ah I see. ok

Reply:
ah I see. ok